src/averageAges.js

```diff
@@ -15,47 +15,72 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(man => man.sex === 'm');
+  const menCentury = century ? men.filter(
+    man => Math.ceil(man.died / 100) === century) : men;
+
+  const averageAge = menCentury.reduce(
+    (sum, man) => sum + (man.died - man.born), 0) / menCentury.length;
+
+  return averageAge;
 }
 
 /**
- * Implement calculateWomenAverageAge function
+* Реализовать функцию расчета womenaverageage
  *
- * Function returns average age of women in array. If `withChildren` is
- * specified then function calculates average age only for women with children
+ * Функция возвращает средний возраст женщин в массиве.Если `witchildren
+ * Указано, затем функционирование вычисляет средний возраст
+ * только для женщин с детьми
  *
- * Hint: To check if a woman has children you should find someone who mention
- * her as mother.
+ * Подсказка: чтобы проверить, есть ли у женщины дети,
+ * вы должны найти кого -то, кто упоминает
+ * ее как мать.
  *
  * @param {object[]} people
  * @param {boolean} withChildren - optional
  *
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(woman => woman.sex === 'f');
+  const womenMothers = women.filter(woman =>
+    people.some(person => person.mother === woman.name));
+  const womenAge = withChildren ? womenMothers : women;
+
+  const averageAge = womenAge.reduce(
+    (sum, woman) => sum + (woman.died - woman.born), 0) / womenAge.length;
+
+  return averageAge;
 }
 
 /**
- * Implement calculateAverageAgeDiff function.
+* Реализовать функцию CalculateAverageadiff.
  *
- * The function returns an average age difference between a child and his or her
- * mother in the array. (A mother's age at child birth)
+ * Функция возвращает среднюю разницу в возрасте между ребенком и его или ей
+ * Мать в массиве.(Возраст матери при рождении ребенка)
  *
- * If `onlyWithSon` is specified then function calculates age difference only
- * for sons and their mothers.
+ * Если указан только `только
+ * Для сыновей и их матерей.
  *
  * @param {object[]} people
  * @param {boolean} onlyWithSon - optional
  *
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon ? people.filter(
+    child => people.some(
+      mother => child.sex === 'm' && mother.name === child.mother))
+    : people.filter(child => people.some(
+      mother => mother.name === child.mother));
```

Comment:
Really hard to read it. Try to refactor this function a little bit.
Here is my suggestion. But maybe you'll have a better variant.
```suggestion
  const children = people.filter(child => people.some(mother => {
    return onlyWithSon
      ? child.sex === 'm' && mother.name === child.mother
      : mother.name === child.mother
  }));
```